token/cli/src/main.rs

```diff
@@ -1295,9 +1295,11 @@ fn command_gc(config: &Config, owner: Pubkey) -> CommandResult {
         }
 
         for (address, (amount, decimals, frozen, close_authority)) in accounts {
-            if address == associated_token_account {
-                // leave the associated token account alone
-                continue;
+            match (address == associated_token_account, close_empty_associated_accounts, amount > 0) {
```

Comment:
actually, i'm just realizing, this should check `total_amount > 0` instead of `amount > 0`. if we have an auxiliary account with some balance and an empty associated account, we shouldn't close the associated account.

this also makes the sanity check actually valuable :smile:

Reply:
absolutely makes sense. Will add that.